### ISSUE TITLE
Feature: Startup option that would ignore any system JVM 

### ISSUE THREAD

**Issue by johnjaylward** [elastic/elasticsearch#55134]
**Describe the feature**:

It would be nice to have a startup option that would tell startup (`elastic-env`) ignore any system JVM installed and to use the bundled one. Maybe an environment variable called something like `ES_USE_BUNDLED_JVM`? Some systems may have legacy JVMs installed in order to support older applications which would be incompatible with ES. I don't expect this to be the case on production machines, but developer machines where they are evaluating the software or just hosting a local copy for development purposes.

Another option would be that during the startup-checks, if the found JVM on JAVA_HOME is too old, try and fallback to the bundled one.

**Comment by jasontedor** [comment]
If you can set an environment variable `ES_USE_BUNDLED_JVM` before executing Elasticsearch like you suggest, can’t you just as easily unset `JAVA_HOME` before executing Elasticsearch? If `JAVA_HOME
` is not set, Elasticsearch will use the bundled JDK.

**Comment by johnjaylward** [comment]
If I was starting it manually I guess I could remember to do that everytime, but that's pretty annoying. If starting it as a service, then no, there isn't a good way to do that with the current scripts

I want to leave JAVA_HOME in tact for other services. It makes much more sense to have variables dedicated to the platform they are effecting.

At system startup the only way to handle this at the moment is to comment out the lines in the `elasticsearch-env` or `elasticsearch-env.bat` that try to use JAVA_HOME first.

**Comment by rjernst** [comment]
Having another environment variable means another switch that can be out of sync. Imagine a user sets JAVA_HOME, and is confused why the bundled jdk is still being used, only to find some other mechanism that is setting this USE_BUNDLED_JVM.

On linux/macos, you can actually set JAVA_HOME to an empty string to get the effect you want, since in bash the `-z` check we use returns true if JAVA_HOME is undefined or empty. I believe making this behavior consistent with windows would net the desired behavior for you, as you could set JAVA_HOME only for your elasticsearch process just before starting (while unsetting is more difficult, and doesn't even appear possible eg for ES as a windows service). @johnjaylward WDYT?

**Comment by johnjaylward** [comment]
While I agree that it's good to have the systems match, I'm not sure how I I would specify an empty `JAVA_HOME` on service startup.

Full use-case:
1. install elastic via the zip file
1. NEW: in the current session set JVM_HOME to "" (powershell: `$Env:JAVA_HOME=""` cmd: `SET "JAVA_HOME="`)
1. perform the setup according to the [setup docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/zip-windows.html)
1. call `bin\elasticsearch-service.bat install`
1. call `elasticsearch-service.bat manager` if any adjustments are needed (startup automatic for example)

Step 2 now gets me installed, however on reboot, will the service will fail to start as the incompatible JVM installed at JAVA_HOME is still set globally, or does the service install store which JVM it was initially configured with?

If the service keeps the JVM setting, I think the solution in the PR works for me. I would note that the installation docs should probably make a note for the option to blank the JAVA_HOME

**Comment by rjernst** [comment]
> Step 2 now gets me installed, however on reboot, will the service will fail to start as the incompatible JVM installed at JAVA_HOME is still set globally, or does the service install store which JVM it was initially configured with?

The windows service handling of JAVA_HOME had been funky, resolving at service start time, but we fixed this since 7.2.0: https://github.com/elastic/elasticsearch/pull/39714

**Comment by johnjaylward** [comment]
Thanks, I believe this fixes the issue for me then!
